libs/tokens/src/state/tokens/userAddedTokensAtom.ts

```diff
@@ -49,6 +49,9 @@ export const removeUserTokensAtom = atom(null, (get, set, tokens: string[]) => {
   const stateCopy = { ...userAddedTokensState[chainId] }
 
   tokens.forEach((token) => {
+    // Important! We need to remove the token from the state using both the original and lowercase address
+    // Because state might be spoiled with mixed case addresses
+    delete stateCopy[token]
```

Comment:
What it here token is already lowercased?
Shouldn't we use `getAddress` https://docs.ethers.org/v5/api/utils/address/#utils-getAddress to checksum it first?

Reply:
Thanks @alfetopito !
Added it as well: https://github.com/cowprotocol/cowswap/pull/5748/commits/c96f5fd7ca28630fb21620d6c72a5b9b2b5de27c